vmdk_plugin/plugin.go

```diff
@@ -249,10 +249,10 @@ func (d *vmdkDriver) Create(r volume.Request) volume.Response {
 	if err != nil {
 		log.WithFields(log.Fields{"name": r.Name,
 			"error": err}).Error("Attach volume failed, removing the volume ")
-		err = d.ops.Remove(r.Name, nil)
-		if err != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Remove volume failed ")
-			return volume.Response{Err: err.Error()}
+		errRm := d.ops.Remove(r.Name, nil)
+		if errRm != nil {
+			log.WithFields(log.Fields{"name": r.Name, "error": errRm}).Error("Remove volume failed ")
+			return volume.Response{Err: errRm.Error()}
```

Comment:
>Its preferable that the error returned on a create error is the create error itself vs. a remove error (which is confusing to the user, when they issued a create command). 

Agree.

>OTOH, how did this cause the empty response from docker to the docker command line. 

err was being reassigned to nil inside the rollback block after the attach error, and at the end we would return the volume response with an error nil since we didn't got any errors from the rollback operations. Which if we had, in any case we would return from there.

Tests in plugin.go:

line 248   //dev, err := d.ops.Attach(r.Name, nil)
line 249   err = fmt.Errorf("STUB")
line 250   dev := []byte("an")

**Without the patch:**
root@vsc01:/tmp/docker-volume-vsphere# docker volume create -d vmdk --name vol$$ -o size=1gb
Error response from daemon: create vol25407: Post http://%2Frun%2Fdocker%2Fplugins%2Fvmdk.sock/VolumeDriver.Create: http: ContentLength=78 with Body length 0

**in log:**
2016-10-08 21:32:38.880136128 +0100 WEST [INFO] Attaching volume and creating filesystem fstype=ext4 name=vol25407
2016-10-08 21:32:38.880268068 +0100 WEST [ERROR] Attach volume failed, removing the volume error=STUB name=vol25407

**With the patch:**
root@vsc01:/tmp/docker-volume-vsphere# docker volume create -d vmdk --name vol$$ -o size=1gb
Error response from daemon: create vol25407: VolumeDriver.Create: STUB

**in log:**
2016-10-08 21:37:47.746007178 +0100 WEST [INFO] Attaching volume and creating filesystem name=vol25407 fstype=ext4
2016-10-08 21:37:47.746070732 +0100 WEST [ERROR] Attach volume failed, removing the volume name=vol25407 error=STUB